tools/didc/src/main.rs

```diff
@@ -115,11 +117,34 @@ fn check_file(env: &mut TypeEnv, file: &Path) -> candid::Result<Option<Type>> {
     check_prog(env, &ast)
 }
 
-fn main() -> Result<(), ExitFailure> {
+use codespan_reporting::diagnostic::{Diagnostic, Label};
+fn as_diagnostic(err: candid::Error) -> Diagnostic<()> {
+    if err.span.start == 0 && err.span.end == 0 {
+        Diagnostic::error().with_message(err.message)
+    } else {
+        Diagnostic::error()
+            .with_message("Syntax error")
+            .with_labels(vec![
+                Label::primary((), err.span.clone()).with_message(err.message)
+            ])
+    }
+}
+
+fn main() -> Result<()> {
+    let writer = StandardStream::stderr(term::termcolor::ColorChoice::Auto);
+    let config = term::Config::default();
+
     match Command::from_args() {
         Command::Check { input } => {
+            let file = SimpleFile::new(input.to_str().unwrap(), std::fs::read_to_string(&input)?);
```

Comment:
Without an actual filemap this doesn't do much. Did you want to merge this experiment, or should this be its own PR?

Reply:
I think it's fine to merge it, as it doesn't touch the main candid crate. 

I actually don't quite like the filemap idea, it seems an overkill to store all files we ever touched in memory, just for reporting a nice error message. But I don't have better ideas neither.